types/node/index.d.ts

```diff
@@ -748,6 +748,11 @@ declare module "http" {
     export interface ServerRequest extends IncomingMessage {
         connection: net.Socket;
     }
+
+    export interface ServerResponseHeaders {
+      [key: string]: number | string | string[];
```

Comment:
I just gave this a try and it works. And it has side effects to `getHeader()` typings because if you pass in a `number` in `setHeader()` or `writeHead()` you get a `number` via `getHeader()` - not a `string`.
Also arrays of numbers are allowed and even arrays with mixture of strings and numbers - or everything else which can be appended to a string...
So in then end we are back to `any` in this area - or we force users to stringify in usercode....